model/src/main/java/uk/gov/hmcts/payment/api/service/PciPalPaymentService.java

```diff
@@ -74,6 +133,79 @@ else if (serviceType.equalsIgnoreCase(SERVICE_TYPE_PROBATE))
         });
     }
 
+    public TelephonyProviderAuthorisationResponse getTelephonyProviderLink(PciPalPaymentRequest pciPalPaymentRequest, TelephonyProviderAuthorisationResponse telephonyProviderAuthorisationResponse, String serviceType, String returnURL) {
+        return withIOExceptionHandling(() -> {
+
+            String flowId = getFlowId(serviceType);
+            LOG.info("flowId: {}   launchURL: {}   viewIdURL: {}   callbackUrl: {}   returnURL: {} ", flowId, launchURL, viewIdURL, callbackUrl, returnURL);
+            HttpPost httpPost = new HttpPost(launchURL);
+            httpPost.addHeader(CONTENT_TYPE, APPLICATION_JSON.toString());
+            httpPost.addHeader(authorizationHeader(telephonyProviderAuthorisationResponse.getAccessToken()));
+            TelephonyProviderLinkIdRequest telephonyProviderLinkIdRequest = TelephonyProviderLinkIdRequest.telephonyProviderLinkIdRequestWith().FlowId(flowId)
+                .InitialValues(TelephonyProviderLinkIdRequest.InitialValues.initialValuesWith()
+                    .amount(new BigDecimal(pciPalPaymentRequest.getOrderAmount()).movePointRight(2).toString())
+                    .callbackURL(callbackUrl)
+                    .returnURL(returnURL)
+                    .orderId(pciPalPaymentRequest.getOrderReference())
+                    .currencyCode("GBP")
+                    .build())
+                    .build();
+
+            Gson gson = new Gson();
+            StringEntity entity = new StringEntity(gson.toJson(telephonyProviderLinkIdRequest));
```

Comment:
If we are already using Jackson ObjectMapper, do we need Gson to convert Obj to Json,
objectMapper.writeValueAsString() should work here

Reply:
Fixed.